module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -201,210 +201,196 @@ namespace vision {
-                        arma::vec3 width_rBCc = ballCentreRay * widthDistance;
-
-                        arma::vec3 rBCc = (width_rBCc);
-
-                        // Attach the measurement to the object
-                        b.measurements.push_back(Ball::Measurement());
-                        b.measurements.back().rBCc       = convert<double, 3, 1>(rBCc);
-                        b.measurements.back().covariance = convert<double, 3>(ball_angular_cov).asDiagonal();
-
-                        // Ball cam space info
-                        b.cone.axis     = convert<double, 3>(ballCentreRay);
-                        b.cone.gradient = result.model.gradient;
-
-                        // Angular positions from the camera
-                        b.visObject.screenAngular = convert<double, 2>(cartesianToSpherical(ballCentreRay).rows(1, 2));
-                        b.visObject.angularSize << getParallaxAngle(left, right, cam), getParallaxAngle(top, base, cam);
-
-                        // Add our points
-                        for (auto& point : result) {
-                            b.edgePoints.push_back(convert<double, 3>(
-                                getCamFromScreen(imageToScreen(point, convert<uint, 2>(image.dimensions)), cam)));
-                        }
-                        b.visObject.timestamp       = NUClear::clock::now();
-                        b.visObject.classifiedImage = const_cast<ClassifiedImage*>(rawImage.get())->shared_from_this();
-
-                        balls->push_back(std::move(b));
-                    }
-
-                    for (auto a = balls->begin(); a != balls->end(); ++a) {
-                        Cone<3> acone(convert<double, 3>(a->cone.axis), a->cone.gradient);
-
-                        for (auto b = a + 1; b != balls->end();) {
-                            Cone<3> bcone(convert<double, 3>(b->cone.axis), b->cone.gradient);
-
-                            // If our balls overlap
-                            if (acone.overlaps(bcone)) {
-                                // Pick the better ball
-                                if (acone.gradient < bcone.gradient) {
-                                    // Throw-out b
-                                    b = balls->erase(b);
-                                }
-                                else {
-                                    a = balls->erase(a);
-
-                                    if (a == b) {
-                                        ++b;
-                                    }
-                                }
-                            }
-                            else {
-                                ++b;
-                            }
-                        }
-                    }
-                    emit(std::move(balls));
-                    lastFrame.time = sensors.timestamp;
-                });
+            .then("Ball Detector",
```

Comment:
There seems to be way more changes in this file than there should be.
What happened?